### DISCUSSION TITLE
Support for @odata.type annotation in POST/PATCH payload for identifying object type

### DISCUSSION THREAD

**Discussion by oartyshko** [odata2ts/odata2ts#255] [Q&A] [answered]
I need to pass the `@odata.type` annotation in the payload of a POST request to identify the type of an object. However, I'm encountering an error when trying to do so:

````
app-error-handler.ts:35 Error: Property [@odata.type] not found (in strict mode)! Known user model props: 
    at QueryObject.js:100:27
    at Array.reduce (<anonymous>)
    at QueryObject.js:87:42
    at Array.map (<anonymous>)
    at Configuration.convertToOData (QueryObject.js:86:31)
    ...
````
It seems that the library is not recognizing the `@odata.type` annotation. Is there any support or configuration needed to enable this annotation in the payload? Or is there another recommended way to achieve the same functionality?
So far I found only one way - overriding generated services so it does not execute `QAction.convertUserParams(params)` method.

**Comment by texttechne** [comment]
Hi @oartyshko,
so if I get you right you're just trying to pass a property in the payload which is not known from the meta model. Well, at least the parameter is not announced for this action, right? Is that correct? Usage examples would be great...

However, the error you receive is a plain TypeScript error: You're passing a model with a property which is not known to the interface. The absolutely most easiest way here is just to `@ts-ignore` that statement. 

`odata2ts` should not interfer with your extra props. It's just that the typing is not correct...

**Reply by oartyshko** [reply]
Hi @texttechne,
 
`@ts-ignore` won't help because it is not a plain TypeScript error. The error occurred here:
https://github.com/odata2ts/odata2ts/blob/main/packages/odata-query-objects/src/QueryObject.ts#L158
I was trying to call a method of `odata2ts` generated service with `@odata.type` annotation inside a payload to create an Entity that contains a complex type.  Here is an example of POST payload for a complex type entity: https://www.odata.org/getting-started/basic-tutorial/#modifyData .
It looks like the `odata2ts` does not support annotations in a POST/PATCH payload or maybe some additional configuration is required. 

I've found this PR https://github.com/odata2ts/odata2ts/pull/108 where `allowUnknownProps` param was added but it is not clear how to use it in the `odata2ts` generated services.

**Reply by texttechne** [reply, marked as answer]
Ah ok, got it now. It's a wrong default setting: Ideal for testing but not for production :see_no_evil:
And I didn't provide any setting for this.

Would you be so kind to create a bug ticket? sth. like: error when sending payloads with custom props...

**Reply by oartyshko** [reply]
I apologize that it took me so long to answer. I've just created an issue:  https://github.com/odata2ts/odata2ts/issues/257
